#! /bin/sh /usr/share/dpatch/dpatch-run
## 99_Werror-format-string.dpatch by Steve Langasek <vorlon@debian.org>
##
## DP: Fix gnulib test suite to work with -Werror=format-string.

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' debian~/gnulib-tests/test-xvasprintf.c debian/gnulib-tests/test-xvasprintf.c
--- debian~/gnulib-tests/test-xvasprintf.c	2012-03-31 21:51:00.000000000 +0000
+++ debian/gnulib-tests/test-xvasprintf.c	2012-05-02 00:01:49.018631894 +0000
@@ -96,8 +96,7 @@
 
   {
     /* Silence gcc warning about zero-length format string.  */
-    const char *empty = "";
-    result = xasprintf (empty);
+    result = xasprintf ("%s", "");
     ASSERT (result != NULL);
     ASSERT (strcmp (result, "") == 0);
     free (result);
